platforms/core-runtime/stdlib-java-extensions/src/main/java/org/gradle/api/NonNullApi.java

```diff
@@ -28,18 +26,23 @@
 /**
  * Marks a type or a whole package as providing a non-null API by default.
  *
- * All parameter and return types are assumed to be {@link Nonnull} unless specifically marked as {@link Nullable}.
+ * All parameter and return types are assumed to be {@link javax.annotation.Nonnull} unless specifically marked as {@link javax.annotation.Nullable}.
  *
  * All types of an annotated package inherit the package rule.
  * Subpackages do not inherit nullability rules and must be annotated.
  *
  * @since 4.2
+ * @deprecated Deprecated in Gradle 9.0 for removal in Gradle 10.
+ * Prefer JSpecify annotations such as {@link org.jspecify.annotations.NullMarked} and {@link org.jspecify.annotations.Nullable}.
+ * Note that you can also still use JSR305 annotations such as {@link javax.annotation.Nonnull} and {@link javax.annotation.Nullable}.
```

Comment:
Do we plan to keep the JSR 305 dependency indefinitely?

Reply:
We need to keep it working for plugins using it on their types.
We can make a decision towards 10.